src/transaction.rs

```diff
@@ -92,6 +92,10 @@ pub struct Transaction {
 
     /// The number of tokens paid for processing and storage of this transaction.
     pub fee: i64,
+    /// Optional user data to be stored in the account
+    /// TODO: This will be a required field for all contract operations including a simple spend.
+    /// `instruction` will be serialized into `userdata` once Budget is its own generic contract.
+    pub userdata: Option<Vec<u8>>,
```

Comment:
Instead of adding this Optional field here, how about adding it as a constructor in Instruction? `CreateUserdata(Vec<u8>)`.  Then the two can coexist until `Instruction` is reduced with just this one mechanism.

Reply:
@garious that will force a bunch of changes throughout the code, basically everywhere we deconstruct Transaction.  And another bunch of changes to remove that constructor again.